common_helper.go

```diff
@@ -102,102 +64,135 @@ func getTenantForScope(scopeuuid string) (tenant, dbError) {
 	if config.GetBool(useCaching) {
 		// acquire a read lock as this cache has 1 writer as well
 		tenantCachelock.RLock()
-		defer tenantCachelock.RUnlock()
 		ten, exists := tenantCache[scopeuuid]
+		tenantCachelock.RUnlock()
+		dbErr := dbError{}
```

Comment:
This is just initializing an empty error as I have a common return statement and depending on whether the if/else flow is executed, the correct value gets populated.